widget/richtext.go

```diff
@@ -225,6 +225,26 @@ func (t *RichText) cachedSegmentVisual(seg RichTextSegment, offset int) fyne.Can
 	return vis
 }
 
+func (t *RichText) cleanVisualCache() {
+	var deletingSegs []RichTextSegment
+	for seg1 := range t.visualCache {
+		found := false
+		for _, seg2 := range t.Segments {
+			if seg1 == seg2 {
+				found = true
+				break
+			}
+		}
+		if !found {
+			// cached segment is not currently in t.Segments, clear it
+			deletingSegs = append(deletingSegs, seg1)
```

Comment:
Does the map change order when we delete from ut? I wonder if we can just delete it directly instead of heap-allocating a slice on each refresh?

Reply:
It does look like it's *probably safe* but the actual spec isn't super explicit about it: https://stackoverflow.com/questions/56048259/is-it-safe-to-remove-dictionary-keys-while-iterating-over-it

Reply:
I added an early return condition so we will skip the allocation and iteration if we don't need to clean anything